test/provisioner/provision.go

```diff
@@ -256,13 +279,34 @@ func (p *CloudAPIAdaptor) Deploy(ctx context.Context, cfg *envconf.Config, props
 		return err
 	}
 
+	log.Info("Installing peerpod-ctrl")
+	cmd = exec.Command("make", "-C", "peerpod-ctrl", "deploy")
+	// Run the deployment from the root src dir
+	cmd.Dir = p.rootSrcDir
+	// Set the KUBECONFIG env var
+	cmd.Env = append(os.Environ(), fmt.Sprintf("KUBECONFIG="+cfg.KubeconfigFile()))
+	stdoutStderr, err = cmd.CombinedOutput()
+	log.Tracef("%v, output: %s", cmd, stdoutStderr)
+	if err != nil {
+		return err
+	}
+
+	// Wait for the peerpod-ctrl deployment to be ready
+	log.Info("Wait for the peerpod-ctrl deployment to be available")
+	if err = wait.For(conditions.New(resources).DeploymentConditionMatch(
+		&appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: "peerpod-ctrl-controller-manager", Namespace: p.namespace}},
+		appsv1.DeploymentAvailable, corev1.ConditionTrue),
+		wait.WithTimeout(time.Minute*5)); err != nil {
+		return err
+	}
+
 	return nil
 }
 
 func (p *CloudAPIAdaptor) DoKustomize(ctx context.Context, cfg *envconf.Config) {
 }
 
-// TODO: convert this into a klient/wait/conditions
+// TODO: convert this into a client/wait/conditions
```

Comment:
`sigs.k8s.io/e2e-framework/klient` is the go package that waits and conditions are under, so I don't think this was a typo

Reply:
Ah this was autocorrect on save :-( .. Need to disable on my new editor setup